modules/OCPP201/OCPP201.cpp

```diff
@@ -1018,4 +1033,53 @@ void OCPP201::process_deauthorized(const int32_t evse_id, const int32_t connecto
     this->process_tx_event_effect(evse_id, tx_event_effect, session_event);
 }
 
+void OCPP201::publish_charging_schedules(const std::vector<ocpp::v201::CompositeSchedule>& composite_schedules) {
+    const auto everest_schedules = conversions::to_everest_charging_schedules(composite_schedules);
+    this->p_ocpp_generic->publish_charging_schedules(everest_schedules);
+}
+
+void OCPP201::set_external_limits(const std::vector<ocpp::v201::CompositeSchedule>& composite_schedules) {
+    const auto start_time = ocpp::DateTime();
+
+    // iterate over all schedules reported by the libocpp to create ExternalLimits
+    // for each connector
+
+    for (const auto& composite_schedule : composite_schedules) {
+        types::energy::ExternalLimits limits;
+        std::vector<types::energy::ScheduleReqEntry> schedule_import;
+
+        for (const auto& period : composite_schedule.chargingSchedulePeriod) {
+            types::energy::ScheduleReqEntry schedule_req_entry;
+            types::energy::LimitsReq limits_req;
+            const auto timestamp = start_time.to_time_point() + std::chrono::seconds(period.startPeriod);
+            schedule_req_entry.timestamp = ocpp::DateTime(timestamp).to_rfc3339();
+            if (composite_schedule.chargingRateUnit == ocpp::v201::ChargingRateUnitEnum::A) {
+                limits_req.ac_max_current_A = period.limit;
+                if (period.numberPhases.has_value()) {
+                    limits_req.ac_max_phase_count = period.numberPhases.value();
+                }
+            } else {
+                limits_req.total_power_W = period.limit;
+            }
+            schedule_req_entry.limits_to_leaves = limits_req;
+            schedule_import.push_back(schedule_req_entry);
```

Comment:
I think this function could be made more readable by moving this code into a `to_schedule_request_entry(period, unit)` helper (possibly in `conversions`?) or a custom constructor. I believe the same could be said for the analogous method in the `OCPP` module.

Reply:
Agreed. We'll target this change in the refactor for both modules in a subsequent PR based on [this issue](https://github.com/EVerest/everest-core/issues/852)